pkg/k8s/watchers/cilium_endpoint_slice_subscriber_test.go

```diff
@@ -0,0 +1,497 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+package watchers
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/cilium/cilium/pkg/endpoint"
+	v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	"github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	slim_metav1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
+	"github.com/cilium/cilium/pkg/k8s/types"
+)
+
+type endpointUpdate struct {
+	oldEp, newEp *types.CiliumEndpoint
+}
+
+func epToString(e *types.CiliumEndpoint) string {
+	if e == nil {
+		return "nil"
+	}
+	return fmt.Sprintf("ep(id: %d)", e.Identity.ID)
+}
+
+func (u endpointUpdate) toString() string {
+	return fmt.Sprintf("(%s, %s)", epToString(u.oldEp), epToString(u.newEp))
+}
+
+func epEqual(e1, e2 *types.CiliumEndpoint) bool {
+	return ((e1 == nil) == (e2 == nil)) && (e1 == nil || e1.Identity.ID == e2.Identity.ID)
+}
+
+func updateEqual(u1, u2 endpointUpdate) bool {
+	return epEqual(u1.oldEp, u2.oldEp) && epEqual(u1.newEp, u2.newEp)
+}
+
+type fakeEPWatcher struct {
+	lastUpdate endpointUpdate
+	lastDelete *types.CiliumEndpoint
+}
+
+func createFakeEPWatcher() *fakeEPWatcher {
+	return &fakeEPWatcher{}
+}
+
+func (fw *fakeEPWatcher) endpointUpdated(oldC, newC *types.CiliumEndpoint) {
+	fw.lastUpdate = endpointUpdate{oldC, newC}
+}
+
+func (fw *fakeEPWatcher) endpointDeleted(c *types.CiliumEndpoint) {
+	fw.lastDelete = c
+}
+
+func (fw *fakeEPWatcher) expectUpdate(u endpointUpdate) (string, bool) {
```

Comment:
assertUpdate or equalUpdate?

same for other similar functions

Reply:
Done